SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -2048,10 +2066,31 @@ void ActorSystem::SendCreateEntityRequest(USpatialActorChannel& ActorChannel, ui
 		   ActorChannel.GetEntityId(), Actor->HasAuthority());
 
 	SpatialGDK::EntityFactory DataFactory(NetDriver, NetDriver->PackageMap, NetDriver->ClassInfoManager, NetDriver->RPCService.Get());
-	TArray<FWorkerComponentData> ComponentDatas = DataFactory.CreateEntityComponents(&ActorChannel, OutBytesWritten);
+	TArray<FWorkerComponentData> ComponentDatas;
+	TArray<FWorkerComponentUpdate> ComponentUpdates;
+	enum EEntityCreationType
+	{
+		CreateNewEntity,
+		PopulateSkeleton,
+	};
+	EEntityCreationType EntityType = ActorChannel.bIsSkeleton ? PopulateSkeleton : CreateNewEntity;
+	if (EntityType == CreateNewEntity)
+	{
+		ComponentDatas = DataFactory.CreateEntityComponents(&ActorChannel, OutBytesWritten);
+	}
+	else if (EntityType == PopulateSkeleton)
+	{
+		const UnrealMetadata StartupActorMetadata = DataFactory.CreateMetadata(*ActorChannel.Actor);
+		const ComponentData StartupActorMetadataComponent(OwningComponentDataPtr(StartupActorMetadata.CreateComponentData().schema_type),
```

Comment:
Should we be trying to replicate this actor before the startup is finished, and conversely, we have all the skeleton entities in view?